### PR TITLE
[Enhancement] Add clockwise argument to ops box_iou_rotated and nms_rotated

### PR DESCRIPTION
**Pull request by zhanggefan**
- add `clockwise` argument to ops `box_iou_rotated` and `nms_rotated`
- add docstrings to clarify the coordination systems used for the representation of rotated bounding boxes

### CODE REVIEW

File: mmcv/ops/box_iou_rotated.py [hunk 1]
```diff
@@ -14,6 +18,46 @@ def box_iou_rotated(bboxes1, bboxes2, mode='iou', aligned=False):
     of bboxes1 and bboxes2, otherwise the ious between each aligned pair of
     bboxes1 and bboxes2.
 
+    coordinates details:
```
Comment (ZwwWayne) [inline mmcv/ops/box_iou_rotated.py:21]: We may use Notes or some other stand block name in sphinx.
See https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#colored-boxes-note-seealso-todo-and-warnings. 

You may also preview the contents here https://mmcv--1592.org.readthedocs.build/en/1592/api.html#mmcv.ops.nms_rotated

File: mmcv/ops/box_iou_rotated.py [hunk 2]
```diff
@@ -14,6 +18,46 @@ def box_iou_rotated(bboxes1, bboxes2, mode='iou', aligned=False):
...
+        the positive direction along x axis is left->right
```
Comment (ZwwWayne) [inline mmcv/ops/box_iou_rotated.py:22]: The operator assumes: 1) the positive direction along x axis is left->right, 2) the positive direction along y axis is top->down, 3) the w border is in parallel with x axis when angle = 0. However, there are 2 xxx.

File: mmcv/ops/box_iou_rotated.py [hunk 3]
```diff
@@ -14,6 +18,46 @@ def box_iou_rotated(bboxes1, bboxes2, mode='iou', aligned=False):
...
+        the positive direction along y axis is top->down
+        the w border is in parallel with x axis when angle = 0
+        there are 2 opposite definitions of the positive angular direction,
+        CW and CCW. MMCV supports both definitions and by default CW.
```
Comment (ZwwWayne) [inline mmcv/ops/box_iou_rotated.py:26]: Need to explain CW and CCW when they are first used.

File: mmcv/ops/box_iou_rotated.py [hunk 4]
```diff
@@ -14,6 +18,46 @@ def box_iou_rotated(bboxes1, bboxes2, mode='iou', aligned=False):
...
+        Please set clockwise=False if you are using the CCW definition.
```
Comment (ZwwWayne) [inline mmcv/ops/box_iou_rotated.py:27]: 'clockwise=False -> ``clockwise=False``'

**Comment by Johnson-Wang** [comment]
Thanks for your contribution. Please have a check at the comments above. By the way, you may refer to http://livesphinx.herokuapp.com/ and paste the docstrings to the online editor to have a real-time check.

**Comment by zhanggefan** [comment]
Got it. Docstrings have been refactored now.

**Review by ZwwWayne — APPROVED** [review]

File: mmcv/ops/box_iou_rotated.py [hunk 5]
```diff
@@ -39,6 +85,11 @@ def box_iou_rotated(bboxes1, bboxes2, mode='iou', aligned=False):
         ious = bboxes1.new_zeros((rows * cols))
     bboxes1 = bboxes1.contiguous()
     bboxes2 = bboxes2.contiguous()
+    if not clockwise:
+        bboxes1 = bboxes1.clone().detach()
```
[the remark this answers is not in the document]
Comment (zhanggefan) [reply mmcv/ops/box_iou_rotated.py]: Code has been changed. Thank you for your kind suggestions!

File: mmcv/ops/nms.py [hunk 6]
```diff
@@ -392,31 +392,38 @@ def nms_match(dets, iou_threshold):
...
 
     Returns:
         tuple: kept dets(boxes and scores) and indice, which is always the
         same data type as the input.
     """
     if dets.shape[0] == 0:
         return dets, None
+    if not clockwise:
+        dets_cw = dets.clone().detach()
+        dets_cw[..., -1] *= -1
```
[the remark this answers is not in the document]
Comment (zhanggefan) [reply mmcv/ops/nms.py]: Code has been changed as above

File: mmcv/ops/nms.py [hunk 7]
```diff
@@ -392,31 +392,39 @@ def nms_match(dets, iou_threshold):
...
 
     Args:
-        boxes (Tensor):  Rotated boxes in shape (N, 5). They are expected to
+        dets (Tensor):  Rotated boxes in shape (N, 5). They are expected to
             be in (x_ctr, y_ctr, width, height, angle_radian) format.
         scores (Tensor): scores in shape (N, ).
         iou_threshold (float): IoU thresh for NMS.
         labels (Tensor): boxes' label in shape (N,).
+        clockwise (bool): flag indicating whether the positive angular
+            orientation is clockwise. default True.
```
Comment (zhouzaida) [inline mmcv/ops/nms.py:409]:
```suggestion
            orientation is clockwise. default True.
            `New in version 1.4.3.`
```

File: mmcv/ops/box_iou_rotated.py [hunk 8]
```diff
@@ -23,6 +115,8 @@ def box_iou_rotated(bboxes1, bboxes2, mode='iou', aligned=False):
             radian.
         mode (str): "iou" (intersection over union) or iof (intersection over
             foreground).
+        clockwise (bool): flag indicating whether the positive angular
+            orientation is clockwise. default True.
```
Comment (zhouzaida) [inline mmcv/ops/box_iou_rotated.py:119]:
```suggestion
            orientation is clockwise. default True.
            `New in version 1.4.3.`
```

**Review by zhouzaida — APPROVED** [review]
